ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/backup/AndroidBackup.kt

```diff
@@ -0,0 +1,73 @@
+package co.electriccoin.zcash.ui.screen.backup
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context
+import androidx.activity.ComponentActivity
+import androidx.activity.viewModels
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import cash.z.ecc.sdk.model.PersistableWallet
+import co.electriccoin.zcash.ui.MainActivity
+import co.electriccoin.zcash.ui.R
+import co.electriccoin.zcash.ui.screen.backup.ext.Saver
+import co.electriccoin.zcash.ui.screen.backup.state.BackupState
+import co.electriccoin.zcash.ui.screen.backup.state.TestChoices
+import co.electriccoin.zcash.ui.screen.backup.view.BackupWallet
+import co.electriccoin.zcash.ui.screen.backup.viewmodel.BackupViewModel
+
+@Composable
+internal fun MainActivity.WrapBackup(
+    persistableWallet: PersistableWallet,
+    onBackupComplete: () -> Unit
+) {
+    WrapBackup(this, persistableWallet, onBackupComplete)
+}
+
+// This layer of indirection allows for activity re-creation tests
+@Composable
+internal fun WrapBackup(
+    activity: ComponentActivity,
+    persistableWallet: PersistableWallet,
+    onBackupComplete: () -> Unit
+) {
+    val backupViewModel by activity.viewModels<BackupViewModel>()
```

Comment:
No change needed here.  I'm looking ahead at what I was thinking how this might evolve.

If we eliminate this ViewModel, then we can alter that pattern a bit by having two files:

AndroidBackup.kt - contains Android-specific injections (e.g. the copy to clipboard method).

StatefulBackup.kt - A new file.  Contains the state retention logic, then delegates to `BackupWallet`.  This ends up being a cross-platform file, since it no longer touches ViewModel.  `BackupWallet` remains "view only" which makes it easy to continue testing the basic view behavior.

Reply:
Nice look ahead, thank you for your clarification.